### DISCUSSION TITLE
Use golang version specified in the go.mod

### DISCUSSION THREAD

**Discussion by aanm** [renovatebot/renovate#20491] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

As mentioned in the [renovate docs](https://docs.renovatebot.com/modules/manager/gomod/#additional-information):
> When Renovate is running using binarySource=docker (such as in the hosted Mend Renovate app) then it will pick the latest compatible version of Go to run, i.e. the latest 1.x release. Even if the go.mod has a version like go 1.14, Renovate will treat it as a ^1.14 constraint and not =1.14.

Is it possible to constrain renovate on using a particular golang version? I have three stable branches and each branch uses a different golang version. I would like to make sure the dependencies are updated accordingly the go version specified in the `go.mod`

### What you tried so far.

I just followed the documentation and I couldn't find a solution for it. I've tried to run renovate and this are the logs for a stable branch that has golang 1.17 installed:

### Relevant debug logs

<details><summary>Logs</summary>

```
"install-tool golang 1.20.1 && go get -d -t ./... && go mod tidy -compat=1.17 && go mod vendor && go mod tidy -compat=1.17 && go mod tidy -compat=1.17
```

</details>

**Comment by rarkins** [comment, marked as answer]
The Go maintainers recommended that we always run using the latest release, due to the fact they maintain backwards compatibility

**Reply by viceice** [reply]
if you want to pin renovate to an older go version then use the constraints renovate option.

**Reply by rarkins** [reply]
I'm not 100% sure it would work, due to our `go-mod-directive` versioning: https://github.com/renovatebot/renovate/blob/main/lib/modules/versioning/go-mod-directive/index.ts

**Reply by paintmeyellow** [reply]
@rarkins So does it work? Need that feature too.

**Reply by rarkins** [reply]
I'm not sure, and don't have plans for experimenting/testing

**Reply by echarrod** [reply]
This doesn't seem to work for us. 
We have:
```json
  "constraints": {
    "go": "1.22"
  },
```
In our renovate config, but when Renovate updated 
[github.com/go-webauthn/webauthn](https://github.com/go-webauthn/webauthn) from `v0.11.1` -\> `v0.11.2`, it also updated the go version from 1.22.3 -> 1.23.0:
![image](https://github.com/user-attachments/assets/a96756c4-6057-4592-b857-94c615fe76d7)

Is there anything we can do to prevent this from happening?

**Reply by rarkins** [reply]
Go versioning implicitly always implies "greater than or equal to" and defaults to the latest compatible version.

If you have a different requirement, please start a new discussion outlining why the the current approach is not ok.
